test/tests/functional/pbs_cray_check_node_exclusivity.py

```diff
@@ -0,0 +1,560 @@
+             ATTR_resv_timezone: tzone,
+             'reserve_start': now + 10,
+             'reserve_end': now + 300}
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        rid_q = rid.split('.')[0]
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid)
+        node = self.server.status(RESV, 'resv_nodes', id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+        self.server.expect(NODE, {'state': 'free'}, id=resv_node)
+        self.logger.info('Waiting 10s for reservation to start')
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=10)
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node)
+        # Submit a job inside the reservation
+        submit_dir = self.du.create_temp_dir(asuser=TEST_USER)
+        a = {ATTR_q: rid_q}
+        j1 = Job(TEST_USER, attrs=a)
+        j1.create_script(self.script)
+        jid1 = self.server.submit(j1, submit_dir=submit_dir)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(NODE, {'state': 'job-exclusive,resv-exclusive'},
+                           id=resv_node)
+        # waith 5 sec for job to end
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node, offset=5, interval=10)
+
+    def test_reservation_request_exclhost(self):
+        """
+        Test Reservation asking for place=exclhost on HostA
+        will have all resources of the vnodes present
+        on HostA assigned to it
+        """
+        # Submit a reservation with place=exclhost
+        now = int(time.time())
+        a = {'Resource_List.select': '1:ncpus=1:vntype=cray_compute',
+             'Resource_List.place': 'exclhost', 'reserve_start': now + 10,
+             'reserve_end': now + 30}
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid)
+        node = self.server.status(RESV, 'resv_nodes', id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+        self.logger.info('Waiting 10s for reservation to start')
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=10)
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node)
+        self.logger.info('Waiting 20s for reservation to finish')
+        self.server.expect(NODE, {'state': 'free'},
+                           id=resv_node, offset=20)
+
+    def test_reservation_request_force_excl(self):
+        """
+        Test Reservation asking for place=force_excl
+        will not get confirmed if node has
+        force_excl/ignore_excl set on it.
+        """
+
+        a = {'sharing': 'ignore_excl'}
```

Comment:
You  need to define the entire vnode if you define anything.  Add cpus and memory